File: backend/app/utils/mood.py
from typing import Literal


Mood = Literal["happy", "encouraging", "sad", "neutral", "excited"]


def mood_from_score(score: int) -> Mood:
    if score >= 90:
        return "excited"
    if score >= 75:
        return "happy"
    if score >= 50:
        return "encouraging"
    return "sad"


def mood_from_text(text: str) -> Mood:
    lower = text.lower()
    if any(key in lower for key in ["great", "excellent", "awesome", "perfect"]):
        return "excited"
    if any(key in lower for key in ["good", "correct", "nice"]):
        return "happy"
    if any(key in lower for key in ["try", "improve", "almost", "close"]):
        return "encouraging"
    if any(key in lower for key in ["incorrect", "wrong", "not quite"]):
        return "sad"
    return "neutral"

File: lib/mood.ts
export type Mood = "happy" | "encouraging" | "sad" | "neutral" | "excited";

export function moodFromScore(score: number): Mood {
  if (score >= 90) return "excited";
  if (score >= 70) return "happy";
  if (score >= 50) return "encouraging";
  if (score >= 30) return "neutral";
  return "sad";
}

export function moodFromText(_text: string): Mood {
  return "encouraging";
}
